vendor/bat-native-ledger/src/bat/ledger/internal/ledger_impl.cc

```diff
@@ -1364,6 +1372,45 @@ void LedgerImpl::GetConfirmationsHistory(
       to_timestamp_seconds, callback);
 }
 
+void LedgerImpl::RefreshPublisherVerifiedStatus(
+    const std::string& publisher_key,
+    ledger::OnRefreshPublisherVerifiedStatusCallback callback) {
+  bat_publishers_->RefreshPublisherVerifiedStatus(publisher_key, callback);
+}
+
+void LedgerImpl::PanelLoadPublisherList(
+    uint64_t window_id,
+    const std::string& publisher_key) {
+  DownloadPublisherList(
+      std::bind(&LedgerImpl::OnPanelPublisherListDownloaded,
+        this, _1, _2, _3, window_id, publisher_key));
+}
+
+void LedgerImpl::OnPanelPublisherListDownloaded(
+    int response_status_code,
+    const std::string& response,
+    const std::map<std::string, std::string>& headers,
+    uint64_t window_id,
+    const std::string& publisher_key) {
+  if (response_status_code != 200) {
+
+  }
+  LoadPublishersListCallback(response_status_code, response, headers);
+  RefreshPublisherVerifiedStatus(publisher_key,
+      std::bind(&LedgerImpl::OnRefreshPublisherVerifiedStatus,
+        this, _1, _2, window_id));
+}
+
+void LedgerImpl::OnRefreshPublisherVerifiedStatus(
+    ledger::Result result,
+    std::unique_ptr<ledger::PublisherInfo> publisher_info,
+    uint64_t window_id) {
+  OnPanelPublisherInfo(
```

Comment:
We should be using callback mechanism to return info to the panel. This way if something is wrong we can tell panel, something like: `Couldn't download list right now, please try again later.`

Reply:
This has been reworked to return in callback. Let me know if this is what you are looking for.

Reply:
Returned api callback as discussed in DM